drf/accounts/api/views.py

```diff
@@ -0,0 +1,38 @@
+from django.contrib.auth import authenticate, get_user_model
+from django.db.models import Q
+
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import permissions
+from rest_framework_jwt.settings import api_settings
+
+jwt_payload_handler = api_settings.JWT_PAYLOAD_HANDLER
+jwt_encode_handler = api_settings.JWT_ENCODE_HANDLER
+jwt_response_payload_handler = api_settings.JWT_RESPONSE_PAYLOAD_HANDLER
+
+User = get_user_model()
+
+
+class LoginView(APIView):
+    permission_classes = [permissions.AllowAny]
+
+    def post(self, request, *args, **kwargs):
+        if request.user.is_authenticated:
+            return Response({'detail': 'You are already authenticated'},
+                            status=400)
+        data = request.data
+        username = data.get('username')
+        password = data.get('password')
+        user = authenticate(username=username, password=password)
+        qs = User.objects.filter(Q(username__iexact=username) |
```

Comment:
You could do something like ```User.objects.get(....)``` with try/catch block. ```qs.count()``` it's really messy and unclear.